test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,809 @@
+                                        ethers.constants.AddressZero
+                                    )
+                            ).to.be.reverted;
+                        }
+                    );
+                });
+            });
+        });
+    });
+
+    describe("#stageProtocolGovernance", () => {
+        it("stages new ProtocolGovernance for commit", async () => {
+            await this.subject
+                .connect(this.admin)
+                .stageProtocolGovernance(this.newProtocolGovernance.address);
+
+            expect(await this.subject.stagedProtocolGovernance()).to.be.equal(
+                this.newProtocolGovernance.address
+            );
+        });
+        it("sets the stagedProtocolGovernanceTimestamp after which #commitStagedProtocolGovernance can be called", async () => {
+            await this.subject
+                .connect(this.admin)
+                .stageProtocolGovernance(this.newProtocolGovernance.address);
+
+            expect(
+                Number(await this.subject.stagedProtocolGovernanceTimestamp()) -
+                    now() -
+                    Number(await this.protocolGovernance.governanceDelay())
+            ).to.be.lessThanOrEqual(1);
+        });
+
+        describe("access control:", () => {
+            it("allowed: ProtocolGovernance Admin", async () => {
+                await this.subject
+                    .connect(this.admin)
+                    .stageProtocolGovernance(
+                        this.newProtocolGovernance.address
+                    );
+            });
+            it("denied: any other address", async () => {
+                let randomAddr = randomAddress();
+                while (randomAddr == this.admin.address) {
+                    randomAddr = randomAddress();
+                }
+                await withSigner(randomAddr, async (s) => {
+                    await expect(
+                        this.subject
+                            .connect(s)
+                            .stageProtocolGovernance(
+                                this.newProtocolGovernance.address
+                            )
+                    ).to.be.reverted;
+                });
+            });
+        });
+
+        describe("edge cases", () => {
+            describe("when new ProtocolGovernance is a zero address", () => {
+                it("does not fail", async () => {
+                    await expect(
+                        this.subject
+                            .connect(this.admin)
+                            .stageProtocolGovernance(
+                                ethers.constants.AddressZero
+                            )
+                    ).to.not.be.reverted;
+                });
+            });
+        });
+    });
+
+    describe("#commitStagedProtocolGovernance", () => {
+        it("commits staged ProtocolGovernance resets staged ProtocolGovernance and ProtocolGovernanceTimestamp", async () => {
```

Comment:
Let's split into 3 cases
commits
resets stagedProtocolGovernance
resets timestamp

Reply:
The stage part can be moved to beforeEach